packages/contracts/contracts/L1/Colosseum.sol

```diff
@@ -231,86 +313,114 @@ contract Colosseum is Initializable, Semver {
+        // TODO(pangssu): call `createTransaction` to guardian
 
-        L2_ORACLE.deleteL2Outputs(challenge.outputIndex);
-        emit ProofCompleted(challengeId, challenge.outputIndex);
-        _closeChallenge(challengeId, challenge);
+        emit Proven(_outputIndex, dstOutputRoot);
     }
 
-    function asserterTimeout() external {
-        uint256 challengeId = latestChallengeId;
-        Types.Challenge storage challenge = challenges[challengeId];
+    /**
+     * @notice Closes the challenge because the challenger timed out.
+     *         The winner is the validator.
+     *
+     * @param _outputIndex Index of L2 checkpoint output.
+     */
+    function challengerTimeout(uint256 _outputIndex) external {
+        Types.Challenge storage challenge = challenges[_outputIndex];
         _validateTurn(challenge, ValidateTurnMode.NORMAL);
-        L2_ORACLE.deleteL2Outputs(challenge.outputIndex);
-        _closeChallenge(challengeId, challenge);
+        _closeChallenge(_outputIndex, challenge);
     }
 
-    function challengerTimeout(uint256 _challengeId) external {
-        Types.Challenge storage challenge = challenges[_challengeId];
-        _validateTurn(challenge, ValidateTurnMode.NORMAL);
-        _closeChallenge(_challengeId, challenge);
+    /**
+     * @notice Approve the challenge to replace output.
+     *
+     * @param _outputIndex Index of L2 checkpoint output.
+     */
+    function approveChallenge(uint256 _outputIndex) external {
+        require(msg.sender == GUARDIAN, "Colosseum: sender is not the guardian");
+
+        Types.Challenge storage challenge = challenges[_outputIndex];
+
+        require(
+            _challengeStatus(challenge) == ChallengeStatus.PROVEN,
+            "Colosseum: this challenge is not validated"
+        );
+
+        L2_ORACLE.replaceL2Output(_outputIndex, challenge.segments[0], challenge.current);
+        _closeChallenge(_outputIndex, challenge);
     }
 
+    /**
+     * @notice Reverts if it's not sender's turn.
+     *
+     * @param _challenge The challenge data.
+     * @param _mode      Turn validation mode.
+     */
     function _validateTurn(Types.Challenge storage _challenge, ValidateTurnMode _mode)
         private
         view
     {
         ChallengeStatus status = _challengeStatus(_challenge);
         require(status != ChallengeStatus.NONE, "Colosseum: challenge not found");
 
-        address expectedSender = _isTimedOut(status) ? _challenge.current : _challenge.next;
+        address expectedSender = _challenge.next;
+        if (status == ChallengeStatus.CHALLENGER_TIMEOUT) {
+            expectedSender = _isChallengerTurn(_challenge.turn)
+                ? _challenge.current
+                : _challenge.next;
+        } else if (status == ChallengeStatus.ASSERTER_TIMEOUT) {
+            expectedSender = _challenge.current;
+        }
         require(expectedSender == msg.sender, "Colosseum: not your turn");
```

Comment:
Please add the comment that "This should not be invoked via a message call, as it may alter the semantic of `msg.sender`, which is used for turn validation."